src/hooks/useEntitlement.ts

```diff
@@ -0,0 +1,51 @@
+import { useQueries } from 'react-query';
+import { useMemo } from 'react';
+
+import type { GetEntitlementsResponse } from '../../types/checkout';
+import type { MediaOffer } from '../../types/media';
+
+import { useConfigStore } from '#src/stores/ConfigStore';
+import { useAccountStore } from '#src/stores/AccountStore';
+import { getEntitlements } from '#src/services/checkout.service';
+
+export type UseEntitlementResult = {
+  isEntitled: boolean;
+  isLoading: boolean;
+  error: unknown | null;
+};
+
+export type UseEntitlement = (mediaOffers?: MediaOffer[], enabled?: boolean) => UseEntitlementResult;
+
+type QueryResult = {
+  responseData?: GetEntitlementsResponse;
+};
+
+const useEntitlement: UseEntitlement = (mediaOffers = [], enabled = true) => {
+  const sandbox = useConfigStore(({ config }) => config?.cleengSandbox);
+  const jwt = useAccountStore(({ auth }) => auth?.jwt);
+
+  const entitlementQueries = useQueries(
+    mediaOffers.map(({ offerId }) => ({
+      queryKey: ['mediaOffer', offerId],
+      queryFn: () => getEntitlements({ offerId: offerId || '' }, sandbox, jwt || ''),
+      enabled: enabled && !!jwt && !!offerId,
+    })),
+  );
+
+  const evaluateEntitlement = (queryResult: QueryResult) => !!queryResult?.responseData?.accessGranted;
+
+  return useMemo(
+    () =>
+      entitlementQueries.reduce<UseEntitlementResult>(
```

Comment:
Not sure I agree with the approach of using reduce here. For example, if their are 3 queries, 1 has an error, 1 is successful, and one hangs for a long time for whatever reason  - this will lead to a result that indicates an error and shows a spinner for a long time.  In reality, we can probably optimize to return isEntitled as soon as we get any 1 query returned without error with accessGranted = true.  Thoughts?

Reply:
Hm, that's correct. I figured we want to return only one object that represents all queries to keep its implementation simple. So any `isLoading` or `error` would work for the total... But you're right that any accessGranted should overrule it. I could filter out the `accessGranted` first and return it if found, and otherwise return the reduced result.
Thanks!

Reply:
I'm not totally sure what to do with the errors. In theory an offerId could be broken and then it should just get ignored. If all of the offers are broken I guess we would just default to saying no access until the queries invalidate and try again. Should we enable retries on the queries?

Reply:
Agreed that an error could probably be ignored since I guess the most probable cause would be older, broken offers. A retry wouldn't make sense in such cases, but perhaps it would for temporal issues? I'd hope these are edge cases though.
(React-query actually retries by default, but I just agreed with you both to take `useQueries` out. 😅 )

Reply:
Can we usequeries without caching but with retries?

Reply:
Yes, or actually just a simple invalidation. I think I should take the user's offers from the CheckoutStore, which updates after a TVOD offer has been purchased. These 'userOffers' (so to say) should be a dependency of useQueries, so I store it within the queryKey, which means it will retrigger after update. It's basically the same as using it as a dependency for a useEffect that would call/recall the entitlement endpoint.

If you're both okay with it, I keep the useQueries, implement all other feedback and showcase what I mean in the next PR.

Reply:
Yes okay, let's evaluate the invalidation from the next PR which includes purchasing the media offers.